diff --git a/extras/wordgrinder.desktop b/extras/wordgrinder.desktop
index 56a593e..783b9c7 100644
--- a/extras/wordgrinder.desktop
+++ b/extras/wordgrinder.desktop
@@ -1,9 +1,12 @@
 [Desktop Entry]
+Type=Application
 Name=WordGrinder
 Comment=Word processor for processing words
-Exec=xwordgrinder %f
+Exec=/usr/bin/invoker --single-instance --type=e /usr/bin/fingerterm -e /usr/bin/wordgrinder %f
 Icon=wordgrinder
-MimeType=application/x-wordgrinder
-Terminal=false
-Type=Application
-Categories=Office;WordProcessor
+MimeType=application/x-wordgrinder;
+Terminal=true
+Categories=Office;WordProcessor;
+
+[X-Sailjail]
+Permissions=Documents
